openedx/core/djangoapps/edx_global_analytics/utils.py

```diff
@@ -0,0 +1,128 @@
+"""
+Helpers for the edX global analytics application.
+"""
+
+import datetime
+import calendar
+
+from django.core.cache import cache
+
+from django.db.models import Count
+from django.db.models import Q
+from student.models import UserProfile
+
+
+def calculate_active_students_amount(query_name, activity_period):
```

Comment:
Also you can use memoize pattern here see [django-memoize](https://pythonhosted.org/django-memoize/) for the reference